CAIPs/caip-draft_cacao.md

```diff
@@ -0,0 +1,191 @@
+
+In the future, we anticipate creating a specialized registry for signature types.
+
+```
+type Signature struct {
+  t String //= "eip191" or "eip1271"
+  m optional SignatureMeta
+  s Bytes
+}
+
+type SignatureMeta struct {
+}
+```
+
+This construction allows a dApp to uniformly request a SIWE signature regardless of the user's account nature. The user's wallet determines if it should use an EOA or a contract account.
+
+### Signature Verification
+
+We reconstruct the EIP4361 payload as follows:
+
+```
+{.p.domain} wants you to sign in with your Ethereum account:
+{.p.iss[address]}
+
+{.p.statement}
+
+URI: {.p.aud}
+Version: {.p.version}
+Chain ID: {.p.iss[chainId]}
+Nonce: {.p.nonce}
+Issued At: {.p.iat}
+Resources:
+- {.p.resources[0]}
+- {.p.resources[1]}
+...
+- {.p.resources[n]}
+```
+
+Signature verification goes according to `t` in `SignatureMeta`:
+- `eip191`: use [EIP-191](https://eips.ethereum.org/EIPS/eip-191),
+- `eip1271`: use [EIP1271](https://eips.ethereum.org/EIPS/eip-1271).
+
+### Serialization
+
+As a proper IPLD object, it can be deterministically serialized using [CBOR](https://ipld.io/docs/codecs/known/dag-cbor/) into bytes.
+Performance is almost as fast as vanilla JSON serialization. For transport purposes we propose that a CACAO is passed inside a base64url-serialized [CAR](https://ipld.io/specs/transport/car/) file,
+with root of the CAR file set to a tip of capability chain. Here and now we use [CARv1](https://ipld.io/specs/transport/car/carv1/) format, as [CARv2](https://ipld.io/specs/transport/car/carv2/) is still being worked on.
+
+We propose, that all the necessary parent CACAOs are passed there as well. This way, even if a referenced CACAO is not yet available over IPFS, both consumer and presenter of CACAO still can access it.
+
+## Rationale
+
+- As a chain-agnostic standard, a capability should identify chain-specific signature methods.
+- While "Sign-in with Ethereum" standardizes payload format, the payload could be extended in future.
+- The standard should be usable for DID-based signing methods as well as blockchain based ones.
+- The format we are creating here should be uniquely serialized as an IPLD object; we expect it to be identified by CID.
+- A capability format described here should allow chaining capabilities together.
+- We should standardize on a url-safe serialization format of a capability chain suitable for well-established non-binary transport protocols.
+
+## Backwards Compatibility
+
+Not applicable.
+
+## Test Cases
+
+Below you could find a CACAO, along with its serialized presentation in CAR file.
+
+CACAO:
+```
+{
+  "h": {
+    "t": "eip4361"
+  },
+  "p": {
+    "aud": "http://localhost:3000",
+    "exp": "2022-02-21T14:06:33.940+03:00",
+    "iat": "2022-02-21T13:06:33.933+03:00",
+    "iss": "did:pkh:eip155:1:0xBAc675C310721717Cd4A37F6cbeA1F081b1C2a07",
+    "nbf": "2022-02-21T13:06:33.940+03:00",
+    "uri": "http://localhost:3000/login",
```

Comment:
Shouldn't this be `domain` and if it is `domain`, we need only the `authority` part.